MetaAudienceNetworkAdapter/src/main/java/com/chartboost/mediation/metaaudiencenetworkadapter/MetaAudienceNetworkAdapter.kt

```
@@ -607,6 +654,102 @@ class MetaAudienceNetworkAdapter : PartnerAdapter {
+                        )
+                    )
+                }
+
+                override fun onAdClicked(ad: Ad?) {
+                    PartnerLogController.log(DID_CLICK)
+                    partnerAdListener.onPartnerAdClicked(
+                        PartnerAd(
+                            ad = ad,
+                            details = emptyMap(),
+                            request = request
+                        )
+                    )
+                }
+
+                override fun onLoggingImpression(ad: Ad?) {
+                    PartnerLogController.log(DID_TRACK_IMPRESSION)
+                    partnerAdListener.onPartnerAdImpression(
+                        PartnerAd(
+                            ad = ad,
+                            details = emptyMap(),
+                            request = request
+                        )
+                    )
+                }
+
+                override fun onRewardedInterstitialCompleted() {
+                    PartnerLogController.log(DID_REWARD)
+                    partnerAdListener.onPartnerAdRewarded(
+                        PartnerAd(
+                            ad = rewardedInterstitialAd,
+                            details = emptyMap(),
+                            request = request
+                        )
+                    )
+                }
+
+                override fun onRewardedInterstitialClosed() {
+                    PartnerLogController.log(DID_DISMISS)
+                    partnerAdListener.onPartnerAdDismissed(
+                        PartnerAd(
+                            ad = rewardedInterstitialAd,
+                            details = emptyMap(),
+                            request = request
+                        ), null
+                    )
+                }
+            }
+
+            rewardedInterstitialAd.loadAd(
+                rewardedInterstitialAd.buildLoadAdConfig()
+                    .withBid(request.adm)
```

should we skip this step if adm is empty?

---

We can't "skip" it - Meta is bidding only so we can submit the `adm` as-is or fail early. Either way there will be an error so it should be fine regardless.